### ISSUE TITLE
object typedef with no property warns with jsdoc/require-property

### ISSUE THREAD

**Issue by dmnsgn** [gajus/eslint-plugin-jsdoc#1082]
## Expected behavior

No warning to be reported for typedef or an option to disable the behaviour.
We might want to document a type object without properties.

## Actual behavior

```bash
warning  Missing JSDoc @property      jsdoc/require-property
```

## ESLint Config

```js
extends: [
      "eslint:recommended",
      "plugin:prettier/recommended",
      "plugin:jsdoc/recommended",
    ],
    plugins: ["eslint-plugin-prettier", "eslint-plugin-jsdoc"],
```

## ESLint sample

```js
/**
 * @typedef {object} MyTypeExtendingAnOption
 */
```

## Environment

- Node version: 20.1.0
- ESLint version 8.40.0
- `eslint-plugin-jsdoc` version: 44.2.3

**Comment by brettz9** [comment]
If you don't want it, just disable the `require-property` rule. That's essentially all the rule does (also for `@namespace`). The rationale for the rule is that it is almost always under-specifying one's docs to not specify specific expected properties.

Closing but feel free to comment further as needed.

**Comment by dmnsgn** [comment]
Thanks for clarifying, I mistakenly thought it was also affecting require-param.

**Comment by brettz9** [comment]
We really should have something comparable for `require-param` which prevents `object` unless it is the root parent of children like:

```js
/**
 * @param {object} cfg
 * @param {string} cfg.str
 */
```
